src/relay/op/type_relations.cc

```diff
@@ -0,0 +1,135 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file type_relations.cc
+ * \brief A set of utilities and common functionality
+ * for type relations.
+ */
+#include <tvm/relay/expr.h>
+#include <tvm/relay/logging.h>
+#include <tvm/relay/op.h>
+#include "../pass/incomplete_type.h"
+
+namespace tvm {
+namespace relay {
+
+TensorType as_ttype(const Type& t) {
+  if (auto tt_node = t.as<TensorTypeNode>()) {
+    return GetRef<TensorType>(tt_node);
+  } else {
+    return TensorType(nullptr);
+  }
+}
+
+// TODO(@jroesch) what size value do we extract?
+int to_int(const tvm::Expr& e) {
+  auto imm = e.as<tvm::ir::IntImm>();
+  CHECK(imm);
+  std::cout << "TYPE: " << imm << imm->type << std::endl;
+  return imm->value;
+}
+
+Array<Type> IdentityRel(const Array<Type>& types, int num_args) {
```

Comment:
why does it need num_args when we can always get it from types?

Reply:
Because the array contains input and output variables, we made it this way to simplify the calling convention between C++ and Python.